manual/src/cmds/ocamldep.etex

```diff
@@ -149,55 +151,63 @@ Display a short usage summary and exit.
 %
 \end{options}
 
-\section{s:ocamldep-makefile}{A typical Makefile}
+\section{s:ocamldep-makefile}{A Makefile}
 
-Here is a template "Makefile" for a OCaml program.
+Here is an example "Makefile" for a OCaml program in order to
+demonstrate the contribution of ocamldep to the compilation model.
+
+Note that it is recommended to use
+higher-level build tools like \href{https://dune.build/}{dune} rather than make.
 
 \begin{verbatim}
 OCAMLC=ocamlc
 OCAMLOPT=ocamlopt
 OCAMLDEP=ocamldep
-INCLUDES=                 # all relevant -I options here
-OCAMLFLAGS=$(INCLUDES)    # add other options for ocamlc here
+INCLUDES=  # all relevant -I options here
+OCAMLFLAGS=$(INCLUDES)	# add other options for ocamlc here
 OCAMLOPTFLAGS=$(INCLUDES) # add other options for ocamlopt here
 
 # prog1 should be compiled to bytecode, and is composed of three
 # units: mod1, mod2 and mod3.
 
-# The list of object files for prog1
+# The list of object files for prog1 in (manually maintained) link order.
```

Comment:
You could add line explaining that the link order maps to the initialization order of modules, and thus different ordering can lead to different program behavior.

Reply:
Thanks, updated. For what it's worth, what surprised me about the original Makefile in the docs was that it wouldn't even build my program, since `ocamlc` and `ocamlopt` both complain if `a.ml` is provided before `b.ml` but depends on `b.ml`. (https://github.com/ocaml/ocaml/issues/12925#issue-2092574257).

Reply:
Note that it is also possile to use `ocamldep -sort` to output one possible ordering compatible with the dependency topological order.

Reply:
Thank you! If you think it's an improvement, I can update the example makefile to use `ocamldep -sort`, while including the caveat about link order affecting behavior.

I started with `dune` so never learned how to write makefiles for OCaml and suspected there was a way to avoid writing down the entire dependency structure in the makefile.